### PR TITLE
glock magazine overlays

### PR DESCRIPTION
**Pull request by SonixApache**
Veerse gave me an idea for a shitpost but it required some work so now we have magazine overlays for glocks instead of flat sprites for each magazine type. I also touched some examine stuff related to some of the other glock shit I did before; iterative design or whatever.
dreamseeker_2021-09-14_00-57-41
extended mag overlay featuring codersprite from the nexis may cry series

:cl:
 * imageadd: Glocks now have magazine sprite overlays, though you probably won't notice it.
 * imageadd: Added a glock sprite by Basil

### CODE REVIEW

File: code/modules/projectiles/guns/projectile.dm [hunk 1]
```diff
@@ -118,6 +125,24 @@
 		to_chat(usr, "<span class='rose'>There is no magazine to remove!</span>")
 
 
+/obj/item/weapon/gun/projectile/proc/mag_overlay()
+	if(stored_magazine)
+		var/mag_sprite = initial(stored_magazine.icon_state)
+		if(!magazine_overlay || magazine_overlay.icon_state != "[mag_sprite]")
```
Comment (DamianX) [inline code/modules/projectiles/guns/projectile.dm]:
```suggestion
		if(!magazine_overlay || magazine_overlay.icon_state != mag_sprite) 
```

File: code/modules/projectiles/guns/projectile.dm [hunk 2]
```diff
@@ -118,6 +125,24 @@
 		to_chat(usr, "<span class='rose'>There is no magazine to remove!</span>")
 
 
+/obj/item/weapon/gun/projectile/proc/mag_overlay()
+	if(stored_magazine)
+		var/mag_sprite = initial(stored_magazine.icon_state)
+		if(!magazine_overlay || magazine_overlay.icon_state != "[mag_sprite]") 
+			overlays -= magazine_overlay
+			magazine_overlay = null
```
Comment (DamianX) [inline code/modules/projectiles/guns/projectile.dm]: you can remove this line

File: code/modules/projectiles/guns/projectile.dm [hunk 3]
```diff
@@ -118,6 +125,24 @@
...
+			var/image/magazine_adjustment = image("icon" = 'icons/obj/gun_part.dmi', "icon_state" = "[mag_sprite]")
```
Comment (DamianX) [inline code/modules/projectiles/guns/projectile.dm]:
```suggestion
			var/image/magazine_adjustment = image("icon" = 'icons/obj/gun_part.dmi', "icon_state" = mag_sprite)
```

File: code/modules/projectiles/guns/projectile/pistol.dm [hunk 4]
```diff
@@ -250,12 +249,47 @@
...
+	desc = "The NT Glock is a cheap, ubiquitous sidearm, produced by a NanoTrasen subsidiary. Uses .380AUTO rounds. Its subcompact frame can fit in your pocket. <br><span class='notice'>This one has rad flame stickers on it!</span>"
+	icon_state = "secglockflame"
+	spawn_mag = FALSE
+	clowned = UNCLOWN
+	w_class = W_CLASS_MEDIUM
+	
+/obj/item/weapon/gun/projectile/glock/flame/New()	
+	..()
+	conversionkit = new /obj/item/gun_part/glock_auto_conversion_kit(src)
+	var/image/auto_overlay = image("icon" = 'icons/obj/gun_part.dmi', "icon_state" = "auto_attach")
```
Comment (DamianX) [inline code/modules/projectiles/guns/projectile/pistol.dm:275]: this is copypasted from the `attackby`. Why not put it in update_icon instead?

File: code/modules/projectiles/guns/projectile/pistol.dm [hunk 5]
```diff
@@ -250,12 +249,47 @@
...
+	magwellmod = mag_type_restricted
+	mag_type_restricted = list()
+	var/obj/item/gun_part/silencer/loudener = new /obj/item/gun_part/silencer/loudencer(src)
+	silenced = loudener
+	var/image/louden_overlay = image("icon" = 'icons/obj/gun_part.dmi', "icon_state" = "[loudener.icon_state]_mounted")
+	louden_overlay.pixel_x += silencer_offset[SILENCER_OFFSET_X]
+	louden_overlay.pixel_y += silencer_offset[SILENCER_OFFSET_Y]
+	overlays += louden_overlay
+	gun_part_overlays += louden_overlay
+	var/obj/item/ammo_storage/magazine/m380auto/fatmag = new /obj/item/ammo_storage/magazine/m380auto/extended(src)
```
Comment (DamianX) [inline code/modules/projectiles/guns/projectile/pistol.dm]:
```suggestion
	var/obj/item/ammo_storage/magazine/m380auto/extended/fatmag = new(src)
```

File: code/modules/projectiles/guns/projectile/pistol.dm (hunk 4 above)
Comment (SonixApache) [reply code/modules/projectiles/guns/projectile/pistol.dm:275]: Attackby isn't being called, and it only needs to be ran once (on attach or removal).

**Pull request merged**